typedoc.json

```diff
@@ -0,0 +1,21 @@
+{
+  "mode": "modules",
+  "out": "./docs/autogen",
+  "json": "./docs/autogen/output.json",
+  "theme": "default",
+  "ignoreCompilerErrors": "true",
+  "experimentalDecorators": "true",
+  "emitDecoratorMetadata": "true",
+  "excludePrivate": "true",
+  "excludeExternals": "true",
+  "includeDeclarations": "true",
+  "target": "ES6",
+  "exclude": "**/+(index.ts|*.spec.ts|*.scss|*.html)",
```

Comment:
We don't necessarily want to exclude `index.ts` because we do define some constructs there. 
(though maybe we shouldn't)